backend/src/Edna/Config/CLA.hs

```diff
@@ -0,0 +1,81 @@
+module Edna.Config.CLA
+  ( EdnaOptions(..)
+  , ednaOpts
+  ) where
+
+import Universum
+
+import qualified Data.Char as C
+import Options.Applicative
+  (Parser, ParserInfo, auto, eitherReader, flag', fullDesc, help, helper, info, long, metavar,
+  option, progDesc, short, strOption, switch)
+
+import Edna.Config.Definition (LoggingConfig(..))
+import Edna.Util (NetworkAddress)
+
+ednaOpts :: ParserInfo EdnaOptions
+ednaOpts = info (ednaOpts' <**> helper) $
+  fullDesc <> progDesc "Edna API server"
+
+data EdnaOptions = EdnaOptions
+  { eoConfig :: Maybe FilePath
+  , eoListen :: Maybe NetworkAddress
+  , eoDocs :: Maybe Bool
+  , eoMaxCon :: Maybe Int
+  , eoLogging :: Maybe LoggingConfig
+  , eoDumpConfig :: Bool
+  } deriving stock (Generic, Show)
+
+
+ednaOpts' :: Parser EdnaOptions
+ednaOpts' = EdnaOptions
+      <$> configPathParser
+      <*> listenParser
+      <*> docsParser
+      <*> maxConnectionsParser
+      <*> loggingParser
+      <*> dumpConfigParser
+
+
+configPathParser :: Parser (Maybe FilePath)
+configPathParser = optional $ strOption $
+  short 'c' <>
+  long "config" <>
+  metavar "EDNA_CONFIG" <>
+  help "Path to configuration file."
+
+listenParser :: Parser (Maybe NetworkAddress)
+listenParser = optional $ option auto $
+  long "listen-addr" <>
+  metavar "EDNA_API_LISTEN" <>
+  help "Edna API listen address (host:port)."
+
+docsParser :: Parser (Maybe Bool)
+docsParser = optional $ flag' True $
+  long "serve-docs" <>
+  help "Whether to serve docs."
+
+maxConnectionsParser :: Parser (Maybe Int)
+maxConnectionsParser = optional $ option auto $
+  long "max-connections" <>
+  metavar "EDNA_DB_MAX_CONNECTIONS" <>
+  help "Edna DB max connections."
+
+loggingParser :: Parser (Maybe LoggingConfig)
+loggingParser = optional $ option (eitherReader loggingParser') $
+  short 'l' <>
+  long "logging" <>
+  metavar "EDNA_LOGGING_LEVEL" <>
+  help "Edna logging level: Dev, Prod or Nothing."
+
+loggingParser' :: String -> Either String LoggingConfig
+loggingParser' p = case map C.toLower p of
+  "dev"     -> Right LogDev
+  "prod"    -> Right LogProd
+  "nothing" -> Right LogNothing
```

Comment:
I propose to put this function next to `LoggingConfig` definition and use it in 3 places:
1. `FromJSON` instance.
2. `loggingParser`
3. `loggingLevelEnv`

Reply:
done